src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,34 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+        } catch (DataProcessingException e) {
```

Comment:
this exception will never be thrown, because it is your custom one, and you do not do that inside a `try` block.
And do not forget to pass an `e` when you rethrow the exception, because we want to know the details of what went wrong.
Same in other places.
```suggestion
        } catch (Exception e) {
```